operator/controller/webhook/introute/sync.py

```diff
@@ -76,16 +79,64 @@ def get_mounts(self) -> List[Mapping]:
         return volume_mounts
 
 
-def _spring_cloud_k8s_config(parent) -> Optional[Mapping]:
+def _get_updated_props_srcs(related_configmaps, props_srcs) -> List:
+    """Replaces each props_src name with the latest ConfigMap based on creationTimestamp and prop_src base name"""
+    updated_props = []
+
+    if not props_srcs or not related_configmaps:
```

Comment:
If no `related_configmaps` are present, then I'm assuming the IntegrationRoute is looking for a ConfigMap that doesn't exist. In that case, does it make sense to return the `props_srcs` as-is and instead let the (cm not found) error bubble up through Kubernetes events for better visibility?

Reply:
I think allowing a `cm not found` error to bubble up (rather than returning an empty props list) also makes sense later on in the method if no matching ConfigMaps are found.

Reply:
If that field is empty, it means there are no configmaps at all in the parent's namespace. 

The input to this function is the prop_srcs list from the ir custom resource and the output is just `name: test-prop-14sl3h5`. If I return the props as-is, then the spring app would never find a configmap with the name `test-props`. Although it might since the labels would still be there. I should note this in the docstring or update the function name.

I do think letting an error bubble up when `related_configmaps` is empty but `props_srcs`is not is a good idea. I think if `props_srcs` is empty it's better to just return an empty list as that would indicate there aren't any changes.